tsconfig.json

```diff
@@ -8,7 +8,7 @@
     "declaration": true,
     "sourceMap": true,
     "outDir": "./dist",
-    "lib": ["es2015"],
+    "lib": ["es2021"],
```

Comment:
I can remember if I did this or you @jenbutongit but if I did it we might need to revert it. 

If you did it. It would be good to know why.

Reply:
so we have access to the latest js apis. If you try to use promise.any() you'll get a lib error. Promise.any is available in node 16

The target is still es5 - so all code is transpiled to es5. Is that your concern?

Technically we should only do that for `client`, though. There's no need to compile down so low for server since it's running node 18.

Reply:
promise.any since it was quite cumbersome for the calling functions to handle promise.allSettled - in reality I think they only care if at least one passed or if they all failed.

Reply:
ah cool, this makes sense 👍